pkg/operator/operator.go

```diff
@@ -745,12 +745,28 @@ func newUWMTaskSpec(targetName string, task tasks.Task) *tasks.TaskSpec {
 	return tasks.NewTaskSpec(UWMTaskPrefix+targetName, task)
 }
 
+func (o *Operator) setUpgradeable(ctx context.Context, status configv1.ConditionStatus, message, reason string) {
+	err := o.client.StatusReporter().SetUpgradeable(ctx, status, message, reason)
+	if err != nil {
+		klog.Errorf("error occurred while setting Upgradeable status: %v", err)
+	}
+}
+
 func (o *Operator) sync(ctx context.Context, key string) error {
-	config, err := o.Config(ctx, key)
+	config, warnings, err := o.Config(ctx, key)
+
+	invalidConfigReason := "InvalidConfiguration"
```

Comment:
hmm shouldn't we try to have different reasons for platform and user-defined configs to help managed service SREs? Like we did for tasks.

Reply:
for now, we've only done that for 4.18 https://issues.redhat.com/browse/OCPBUGS-33863, we could backport that, I can take care or that in another PR.

Reply:
sounds good